src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,288 @@
+      for_account: tippingContractAddresses?.value?.tippingV2?.replace('ct_', 'ak_'),
+    });
+
+    const allowanceAmount = decodedResult !== undefined
+      ? new BigNumber(decodedResult)
+        .multipliedBy(-1)
+        .plus(toShiftedBigNumber(amount, selectedToken?.decimals!))
+        .toNumber()
+      : toShiftedBigNumber(amount, selectedToken?.decimals!).toFixed();
+
+    const getContractFunction = (tokenContract.methods as any)[
+      decodedResult !== undefined ? 'change_allowance' : 'create_allowance'
+    ];
+
+    return getContractFunction(
+      tippingContractAddresses.value?.tippingV2?.replace('ct_', 'ak_'),
+      allowanceAmount,
+    );
+  }
+
+  async function getContractTokenPairs(
+    address: Encoded.ContractAddress,
+  ): Promise<Record<string, IToken>> {
+    try {
+      const aeSdk = await getAeSdk();
+      const account = getLastActiveProtocolAccount(PROTOCOL_AETERNITY);
+      const tokenContract = await aeSdk.initializeContract({
+        aci: AedexV2PairACI,
+        address,
+      });
+
+      const [
+        { decodedResult: balances },
+        { decodedResult: balance },
+        { decodedResult: token0 },
+        { decodedResult: token1 },
+        { decodedResult: reserves },
+        { decodedResult: totalSupply },
+      ] = await Promise.all([
+        tokenContract.balances(),
+        tokenContract.balance(account?.address),
+        tokenContract.token0(),
+        tokenContract.token1(),
+        tokenContract.get_reserves(),
+        tokenContract.total_supply(),
+      ]);
+
+      return {
+        token0: {
+          ...availableTokens.value?.[token0],
+          amount: calculateSupplyAmount(
+            balance,
+            totalSupply,
+            reserves.reserve0,
+          ),
+        },
+        token1: {
+          ...availableTokens.value?.[token1],
+          amount: calculateSupplyAmount(
+            balance,
+            totalSupply,
+            reserves.reserve1,
+          ),
+        },
+        totalSupply,
+        balance,
+        balances,
+      };
+    } catch (error) {
+      return {};
+    }
+  }
+
+  async function transfer(
+    address: string,
+    toAccount: any,
+    amount: any,
+    option: any,
+  ) {
+    const aeSdk = await getAeSdk();
+    const tokenContract = await aeSdk.initializeContract({
+      aci: FungibleTokenFullInterfaceACI,
+      address: address as any,
+    });
+    return tokenContract.transfer(toAccount, amount.toFixed(), option);
+  }
+
+  async function burnTriggerPoS(
+    address: string,
+    posAddress: any,
+    invoiceId: any,
+    amount: any,
+    option: any,
+  ) {
+    const aeSdk = await getAeSdk();
+    const tokenContract = await aeSdk.initializeContract({
+      aci: ZeitTokenACI,
+      address: address as any,
+    });
+    return tokenContract.burn_trigger_pos(
+      amount.toFixed(), posAddress, invoiceId, option,
+    );
+  }
+
+  watch(activeNetwork, async (network, oldNetwork) => {
```

Comment:
Previous watcher was initiated at the app runtime and was present all the time. This one will be removed each time components using the composable are unmounted. So maybe it is better to use `onNetworkChange` present in networks composable?